x-pack/plugins/fleet/server/services/artifacts/artifacts.ts

```diff
@@ -82,6 +84,50 @@ export const createArtifact = async (
   return esSearchHitToArtifact({ _id: id, _source: newArtifactData });
 };
 
+export const bulkCreateArtifacts = async (
+  esClient: ElasticsearchClient,
+  artifacts: NewArtifact[],
+  refresh = false
+): Promise<{ artifacts?: Artifact[]; errors?: Error[] }> => {
+  const ids = artifacts.map((artifact) => uniqueIdFromArtifact(artifact));
+  const newArtifactsData = artifacts.map((artifact) =>
+    newArtifactToElasticsearchProperties(artifact)
+  );
+
+  const body = ids
+    .map((id, index) => [
+      {
+        create: {
+          _id: id,
+        },
+      },
+      newArtifactsData[index],
+    ])
+    .flatMap((arr) => arr);
```

Comment:
nit: instead of `flatMap((arr) => arr)`, `flat()` can be used, or even better, the `map` above can be changed to a `flatMap` like this:

```typescript
  const body = ids
    .flatMap((id, index) => [
      {
        create: {
          _id: id,
        },
      },
      newArtifactsData[index],
    ]);
```